src/StartupScripts/OsServiceScriptManager.ts

```diff
@@ -0,0 +1,44 @@
+import { StartupManager } from './StartupManager';
+import { StartupOptions } from './StartupOptions';
+import service from 'os-service';
+import { promisify } from 'util';
+
+export class OsServiceScriptManager implements StartupManager {
+  install(opts: StartupOptions): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      service.add(
+        opts.serviceName,
+        {
+          programArgs: opts.exeArgs,
+          programPath: opts.exePath
+        },
+        (error: Error) => {
+          if (error) {
+            return reject(error);
+          }
+          resolve();
+        }
+      );
+    });
+  }
+
+  run(): Promise<void> {
+    return Promise.resolve(service.run(() => this.stop(0)));
```

Comment:
We have to kill a process and close connection with a bus while stopping.

That's why I proposed to accept a handler via options:

```ts
class OsServiceScriptManager {
  constructor(private readonly options?: { onStop?: (service: StartupManager) => unknown }) {}
}
```

Reply:
stop will send SIGTERM isnt it?
```  process.on('SIGTERM', stop).on('SIGINT', stop).on('SIGHUP', stop);

Reply:
`os-service` will terminate a process by calling `process.exit(code)`. Since `SIGTERM` is not supported on Windows, we should handle a stop manually to free resources properly